### PR TITLE
demo: improve popconfirm/popover/tootlip placement demo code style

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

https://ant.design/components/popover-cn#popover-demo-placement
这类 placment 相关的 demo 有很多样式细节和代码冗余问题，整体重构统一一下。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    --       |
| 🇨🇳 Chinese |   --        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 7ff63e0</samp>

Updated the example code for the `Popconfirm`, `Popover`, and `Tooltip` components to use the `ConfigProvider` component and logical CSS properties for better support of different writing modes. Removed unnecessary `onConfirm` props and `<style>` tags. Added `arrow` props to demonstrate arrow options.

### <samp>🤖 Generated by Copilot at 7ff63e0</samp>

*  Removed `onConfirm` prop from `Popconfirm` components, as it was redundant and unnecessary ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L28),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L38),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L60),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L70),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L92),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L102),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L124),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L134))
*  Changed `marginLeft` style to `marginInlineStart` style, as it was more compatible with different writing modes and directions ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L45-R47),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L77-R76),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L109-R105),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-11aa2afb57b060eea255d1b751e47cecc5f2787ba089076ae10227e7c6cf8cbdL55-R67),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-11aa2afb57b060eea255d1b751e47cecc5f2787ba089076ae10227e7c6cf8cbdL77-R89),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-cca0d1a8008c4edb1206b72cbb7fa044d012fa8068fa3b5013f98002e8b8b912L13-R21),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-cca0d1a8008c4edb1206b72cbb7fa044d012fa8068fa3b5013f98002e8b8b912L39-R43),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-cca0d1a8008c4edb1206b72cbb7fa044d012fa8068fa3b5013f98002e8b8b912L50-R54),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL41-R56),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL76-R81),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL94-R99),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-e536d7a6a9684c924d2d6998d5240ce2a902c1760ca49cd1228ddee2f54cc9c6L52-R68),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-e536d7a6a9684c924d2d6998d5240ce2a902c1760ca49cd1228ddee2f54cc9c6L69-R74))
*  Changed `float` and `clear` styles to `inline-start`, as they were more consistent with the inline axis ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-11aa2afb57b060eea255d1b751e47cecc5f2787ba089076ae10227e7c6cf8cbdL44-R56),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-11aa2afb57b060eea255d1b751e47cecc5f2787ba089076ae10227e7c6cf8cbdL66-R78),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL57-R62),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-e536d7a6a9684c924d2d6998d5240ce2a902c1760ca49cd1228ddee2f54cc9c6L33-R49))
*  Added `ConfigProvider` component to wrap the example code and provide some common button styles, such as width and margin, using the `button` prop ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L141-R130),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-11aa2afb57b060eea255d1b751e47cecc5f2787ba089076ae10227e7c6cf8cbdL12-R45),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-11aa2afb57b060eea255d1b751e47cecc5f2787ba089076ae10227e7c6cf8cbdL77-R89),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-cca0d1a8008c4edb1206b72cbb7fa044d012fa8068fa3b5013f98002e8b8b912L13-R21),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-cca0d1a8008c4edb1206b72cbb7fa044d012fa8068fa3b5013f98002e8b8b912L61-R65),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL34-R32),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL102-R116),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-e536d7a6a9684c924d2d6998d5240ce2a902c1760ca49cd1228ddee2f54cc9c6L76-R90))
*  Changed `buttonWidth` constant from `70` to `80`, as it was more consistent with the default button width in the ant design library ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-11aa2afb57b060eea255d1b751e47cecc5f2787ba089076ae10227e7c6cf8cbdL12-R45),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-cca0d1a8008c4edb1206b72cbb7fa044d012fa8068fa3b5013f98002e8b8b912L13-R21))
*  Replaced `showArrow` and `arrowAtCenter` states with a single `arrow` state, which can have three possible values: `'Show'`, `'Hide'`, or `'Center'` ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-11aa2afb57b060eea255d1b751e47cecc5f2787ba089076ae10227e7c6cf8cbdL12-R45))
*  Changed `value` prop of `Segmented` component from `showArrow` to `arrow`, to match the new state ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL34-R32))
*  Used `useMemo` hook to memoize the `mergedArrow` variable, which passes the `arrow` prop to the `Popover` components ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-11aa2afb57b060eea255d1b751e47cecc5f2787ba089076ae10227e7c6cf8cbdL12-R45))
*  Removed `message` component from the import statement, as it was no longer used in the example code ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-3462134d6ee9e24a66e6a52ac3ae5dcfe44cb1cf1e983c0dab46e9f5983edb72L2-R18))
*  Removed `Divider` component from the import statement, as it was no longer used in the example code ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-11aa2afb57b060eea255d1b751e47cecc5f2787ba089076ae10227e7c6cf8cbdL2-R2),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL2-R8))
*  Removed `<style>` tag from the `components/popover/demo/arrow.md` and `components/popover/demo/placement.md` files, as it was no longer needed ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-ddee156fa9afec3a5ad15ec92c74de00b0c439fec0bc25b4fbc46f66181af4a5L8-L27),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-f78a1b91192a14043ebbffac0ca08c5ff105d9abfe9ea553efa9ddad08019ab7L8-L22))
*  Removed `btnProps` constant and prop from the `components/tooltip/demo/arrow.tsx` and `components/tooltip/demo/placement.tsx` files, as it was no longer needed ([link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL2-R8),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL41-R56),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-9d8282bfef88783426ca34cf677d55aa7d0823b54db0e0ccb2d256262b0524abL102-R116),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-e536d7a6a9684c924d2d6998d5240ce2a902c1760ca49cd1228ddee2f54cc9c6L2-R30),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-e536d7a6a9684c924d2d6998d5240ce2a902c1760ca49cd1228ddee2f54cc9c6L33-R49),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-e536d7a6a9684c924d2d6998d5240ce2a902c1760ca49cd1228ddee2f54cc9c6L52-R68),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-e536d7a6a9684c924d2d6998d5240ce2a902c1760ca49cd1228ddee2f54cc9c6L69-R74),[link](https://github.com/ant-design/ant-design/pull/45581/files?diff=unified&w=0#diff-e536d7a6a9684c924d2d6998d5240ce2a902c1760ca49cd1228ddee2f54cc9c6L76-R90))

### CODE REVIEW

**Comment by afc163** [comment]
+1 Needed

**Review by li-jia-nan — APPROVED** [review]

File: components/popover/demo/arrow.tsx [hunk 1]
```diff
@@ -9,72 +9,86 @@ const content = (
   </div>
 );
 
-const buttonWidth = 70;
+const buttonWidth = 80;
 
 const App: React.FC = () => {
-  const [showArrow, setShowArrow] = useState(true);
-  const [arrowAtCenter, setArrowAtCenter] = useState(false);
+  const options = ['Show', 'Hide', 'Center'];
```
Comment (li-jia-nan) [inline components/popover/demo/arrow.tsx:15]: 这个是常量的话，可以放到组件外面去吧

File: components/popover/demo/arrow.tsx [hunk 2]
```diff
@@ -9,72 +9,86 @@ const content = (
...
+        style: { width: buttonWidth, margin: 4 },
+      }}
+    >
       <Segmented
-        options={['Show', 'Hide', 'Center']}
-        onChange={(val) => {
-          setShowArrow(val !== 'Hide');
-          setArrowAtCenter(val === 'Center');
+        options={options}
+        onChange={(val: string) => {
```
Comment (li-jia-nan) [inline components/popover/demo/arrow.tsx:40]:
```suggestion
        onChange={setArrow}
```

Comment (li-jia-nan) [reply components/popover/demo/arrow.tsx:40]: 可以简化一下

**Review by kiner-tang — APPROVED** [review]

File: components/popover/demo/arrow.tsx (hunk 1 above)
Comment (afc163) [reply components/popover/demo/arrow.tsx:15]: 对，不过放里面问题也不大。

File: components/popover/demo/arrow.tsx (hunk 2 above)
Comment (afc163) [reply components/popover/demo/arrow.tsx:40]: ts 写起来比较麻烦。

File: components/popover/demo/arrow.tsx (hunk 1 above)
Comment (afc163) [reply components/popover/demo/arrow.tsx:15]: 发现没必要提变量：https://github.com/ant-design/ant-design/pull/45585

**Pull request merged**